src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -0,0 +1,82 @@
+package com.rtb.projectmanagementtool.task;
+
+import java.util.HashSet;
+
+/** Class controlling the TaskData object. */
+public final class TaskController {
+
+  private TaskData taskData;
```

Comment:
Sorry to clarify, TaskController doesn't necessarily have to map to only a single TaskData.  Your TaskController can (and normally would) have a list/map of TaskDatas.

Think of the controller as the thing that does the operations on one or more tasks, specifically for example load/save/add/delete tasks.  So most probably this will also hold all of your Datastore load/save logic.